### ISSUE TITLE
[Error: width and height must be > 0] On image compression

### ISSUE THREAD

**Issue by wwessel01** [numandev1/react-native-compressor#112]
### Current behavior

When trying to compress an image from URI or base64 an error is thrown.
`[Error: width and height must be > 0]`

### Expected behavior

To be able to compress an image using base64 and throw a more elaborate error.

### Platform

- [X] Android
- [ ] iOS (Not tested)

### Code

This will throw the error
```
try {
    const compressedSource = await compressFile(
        CompressFileType.IMAGE,
        'https://picsum.photos/500'
    );
    console.log(compressedSource);
} catch (error) {
    console.error(error);
}
```

And this is where the compression is handled
```
import { Image, Video, Audio } from 'react-native-compressor';
import { CompressFileType } from '../../enums/CompressFileType';

const compressFile = async (fileType: CompressFileType, value: string) => {
  console.log('Compressing ' + fileType + ' with value ' + value);
  switch (fileType) {
    case CompressFileType.IMAGE:
      console.log('IMAGE');
      const imageResult = await Image.compress(value, {
        compressionMethod: 'auto',
        // input: 'base64',
        maxHeight: 50,
        maxWidth: 50,
      });
      return imageResult;
    case CompressFileType.VIDEO:
      console.log('VIDEO');
      const videoResult = await Video.compress(value, {
        compressionMethod: 'auto',
      });
      return videoResult;
    case CompressFileType.AUDIO:
      console.log('AUDIO');
      const audioResult = await Audio.compress(value, {
        quality: 'medium',
      });
      return audioResult;
    default:
      throw new Error(`Unknown file type '${fileType}'`);
  }
};

export { compressFile };

```

**Comment by saif-o99** [comment]
same issue here

**Comment by kristijantomic** [comment]
On Anroid same error width and height must be > 0, while on iOS promise never resolve, I Tried example code with auto compression.


```
System:
    OS: macOS 12.3.1
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 932.83 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.2 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.1.2 - /usr/local/bin/npm
    Watchman: 2022.03.21.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /Users/kristijan/.rbenv/shims/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 23, 28, 29, 30, 31
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 31.0.0, 31.0.0
      System Images: android-27 | Google APIs Intel x86 Atom, android-29 | Google Play Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
      Android NDK: 22.1.7171670
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8309675
    Xcode: 13.4/13F17a - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_332 - /usr/bin/javac
  npmPackages:
    react: 17.0.2 => 17.0.2
    react-native: 0.68.2 => 0.68.2
    react-native-macos: Not Found
```

**Comment by brianso123** [comment]
found a work around by removing `compressionMethod: 'auto'`

**Comment by JosMMV** [comment]
I had this issue working with URI returned by react-native-document-picker, I solved it by using react-native-fs copying the file into a temporary path, like this:

```javascript
import { Image } from 'react-native-compressor'
import RNFS from 'react-native-fs'

...

const tempPath = RNFS.TemporaryDirectoryPath + '/' + fileName

await RNFS.copyFile(uri, tempPath)
const compressedImage = await Image.compress(tempPath, {
	compressionMethod: 'auto',
})
```

**Comment by anhnch** [comment]
For me the error occurs if I compress base64 string on Android

**Comment by pontustengroth** [comment]
Anyone find a solution to this?

**Comment by pontustengroth** [comment]
Solved it by providing a file:// path instead of a content:// path

**Comment by vineethsagar** [comment]
@pontustengroth how did you convert from content:// to file:// ?

**Comment by pontustengroth** [comment]
> @pontustengroth how did you convert from content:// to file:// ?

```
const uri = FileSystem.cacheDirectory + `file-${makeid(16)}`;
await FileSystem.copyAsync({
  from: "content://path",
  to: uri,
});
```

Then sending uri for compression

**Comment by dmsierra11** [comment]
The only thing that has worked for me is removing `compressionMethod: 'auto'`, but that disables the compression, so is not really a solution. I have tried other suggestions like saving the file in a temp file and adding `file://` prefix, but no luck. Are the maintainers of this library looking into this?

**Comment by numandev1** [comment]
🙏 sorry for a long ago I was not maintaining due to the huge office workload, hopefully, I am going to start fixing bugs on a weekend basis,
I will fix at least 2 issues on the weekend, for priority bugs you can join the discord channel https://discord.gg/dtYzk8sp where we can discuss which bug is most important

**Comment by numandev1** [comment]
@dmsierra11 @wwessel01 @kristijantomic @saif-o99 should I add support for compressing images from the server like this `https://picsum.photos/500`👀

**Comment by suleymanbasbug** [comment]
> @dmsierra11 @wwessel01 @kristijantomic @saif-o99 should I add support for compressing images from the server like this `https://picsum.photos/500`👀

This is what i require.

**Comment by numandev1** [comment]
@suleymanbasbug @dmsierra11 @kristijantomic @saif-o99 released in `1.7.1`, can you check compression for remote images/videos?

**Comment by noobling** [comment]
I am getting this again for the following path: `file:///data/user/0/com.vendor.example/cache/Camera/6f489980-475c-4924-8aa3-8623fe93ecb6.mp4` is it because it is a video? 

I am on 1.8.24

**Comment by mgmskumara** [comment]
I got this error for base64, i used file , then it works

**Comment by Gul7333** [comment]
Helo bhai... Width and height should be > 0 error AA rha hai... 
Yeh batao k file ka path kis Tarah Dena hai...

**Comment by MadeByRaymond** [comment]
Here is how I solved it, I don't know if it is the same issue. 

When I picked a file to use it returned a uri with a "content://" prefixed on Andriod. It returns `ph://` on iOS, but keep in mind this issue was only happening on Android. 

Using `getRealPath`  i.e:  `const realURIPath = await Compressor.getRealPath(uri, 'image');`, it returned a path for me, `file://file:///storage/emulated/0/Xender/image/IMG_1018.JPG`. This was the path to the image I was trying to compress. 

> _PS: the reason I had to get the real path first is because on iOS, working with `ph://`, the compressor kept throwing that it wasn't a valid URL so opted to get the real path first. That is how I landed here._
**...anyway, back to my explanation.**

I noticed what was being returned had two `file://` prefixes. That seemed off so I did this 
`const result = await Compressor.Image.compress(realURIPath.replaceAll('file://file://', 'file://'));`

This solved the bug for me. 

If I am to draw a conclusion, I will say it it because react-native-compressor can not resolve the path well so it throws this error. Correct me if I am wrong .
